sql/src/main/java/io/crate/protocols/postgres/SimplePortal.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.postgres;
+
+import io.crate.Constants;
+import io.crate.action.sql.ResultReceiver;
+import io.crate.action.sql.RowReceiverToResultReceiver;
+import io.crate.analyze.Analysis;
+import io.crate.analyze.Analyzer;
+import io.crate.analyze.ParameterContext;
+import io.crate.analyze.symbol.Field;
+import io.crate.analyze.symbol.Symbols;
+import io.crate.concurrent.CompletionListener;
+import io.crate.core.collections.Row;
+import io.crate.exceptions.Exceptions;
+import io.crate.executor.Executor;
+import io.crate.executor.transport.kill.KillJobsRequest;
+import io.crate.executor.transport.kill.KillResponse;
+import io.crate.executor.transport.kill.TransportKillJobsNodeAction;
+import io.crate.operation.collect.StatsTables;
+import io.crate.operation.projectors.ResumeHandle;
+import io.crate.planner.Plan;
+import io.crate.planner.Planner;
+import io.crate.sql.tree.Statement;
+import io.crate.types.DataType;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static io.crate.action.sql.SQLBulkRequest.EMPTY_BULK_ARGS;
+
+public class SimplePortal extends AbstractPortal {
+
+    private final static ESLogger LOGGER = Loggers.getLogger(SimplePortal.class);
+
+    private String query;
+    private Statement statement;
+    private Analysis analysis;
+    private Plan plan;
+    @Nullable
+    private FormatCodes.FormatCode[] resultFormatCodes;
+    private List<? extends DataType> outputTypes;
+    private ResultReceiver resultReceiver;
+    private int maxRows = 0;
+    private final List<Object> params = new ArrayList<>();
```

Comment:
I think there is no need to create a new list here and add to it. Instead take the params that are passed into the `bind` method.